doc/en/how-to/tmp_path.rst

```diff
@@ -133,27 +133,47 @@ API for details.
 Temporary directory location and retention
 ------------------------------------------
 
-Temporary directories are by default created as sub-directories of
-the system temporary directory.  The base name will be ``pytest-NUM`` where
-``NUM`` will be incremented with each test run.
-By default, entries older than 3 temporary directories will be removed.
-This behavior can be configured with :confval:`tmp_path_retention_count` and
-:confval:`tmp_path_retention_policy`.
+The temporary directories,
+as returned by the :fixture:`tmp_path` and (now deprecated) :fixture:`tmpdir` fixtures,
+are automatically created under a base temporary directory,
+in a structure that depends on the ``--basetemp`` option:
 
-Using the ``--basetemp``
-option will remove the directory before every run, effectively meaning the temporary directories
-of only the most recent run will be kept.
+- By default (when the ``--basetemp`` option is not set),
+  the temporary directories will follow this this template:
 
-You can override the default temporary directory setting like this:
+  .. code-block:: text
 
-.. code-block:: bash
+      {temproot}/pytest-of-{user}/pytest-{num}/{testname}/
 
-    pytest --basetemp=mydir
+  where:
 
-.. warning::
+  - ``{temproot}`` is the system temporary directory
+    as determined by :py:func:`tempfile.gettempdir`.
+    It can be overridden by the ``PYTEST_DEBUG_TEMPROOT`` environment variable.
```

Comment:
We might as well document this in `reference.rst`, and reference it from here using `:envvar:PYTEST_DEBUG_TEMPROOT`.